vpr/src/pack/re_cluster_util.h

```diff
@@ -0,0 +1,80 @@
+#ifndef RE_CLUSTER_UTIL_H
+#define RE_CLUSTER_UTIL_H
+
+#include "clustered_netlist_fwd.h"
+#include "clustered_netlist_utils.h"
+#include "atom_netlist_fwd.h"
+#include "globals.h"
+#include "pack_types.h"
+#include "cluster_util.h"
+/**
+ * @file
+ * @brief This files defines some helper functions for the re-clustering
+ *
```

Comment:
Maybe add some context to this comment - why would we want to re-do clusters, what situations would that be beneficial in?